docs/modelserving/v1beta1/transformer/collocation/README.md

```diff
@@ -0,0 +1,123 @@
+2. Your cluster's Istio Ingress gateway must be [network accessible](https://istio.io/latest/docs/tasks/traffic-management/ingress/ingress-control/).
+3. You can find the [code samples](https://github.com/kserve/kserve/tree/master/docs/samples/v1beta1/transformer/collocation) on kserve repository.
+
+## Deploy the InferenceService
+
+Since, the predictor and the transformer are in the same pod, they need to listen on different ports to avoid conflict. `Transformer` is configured to listen on port 8000 and 8081 
+while, `Predictor` listens on port 8080 and 8082. `Transformer` calls `Predictor` on port 8082 via local socket. 
+Deploy the `Inferenceservice` using the below command.
+
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: serving.kserve.io/v1beta1
+kind: InferenceService
+metadata:
+  name: custom-transformer-collocation
+spec:
+  predictor:
+    containers:
+      - name: kserve-container
+        image: kserve/custom-model-grpc:latest
+        args:
+          - --model_name=custom-model
+          - --grpc_port=8082
+          - --http_port=8080
+
+      - image: kserve/image-transformer:latest
+        name: transformer-container    # Do not change the container name
+        args:
+          - --model_name=custom-model
+          - --protocol=grpc-v2
+          - --http_port=8000
+          - --grpc_port=8081
+          - --predictor_host=localhost:8082
+        ports:
+          - containerPort: 8000
+            protocol: TCP
+EOF
+```
+!!! success "Expected output"
+    ```{ .bash .no-copy }
+    $ inferenceservice.serving.kserve.io/custom-transformer-collocation created
+    ```
+!!! Warning
+    Always use the transformer container name as `transformer-container`. Otherwise, the model volume is not mounted to the transformer container which may result in an error.
+## Check InferenceService status
+```bash
+kubectl get isvc custom-transformer-collocation
+```
+!!! success "Expected output"
+    ```{ .bash .no-copy }
+    NAME                             URL                                                         READY   PREV   LATEST   PREVROLLEDOUTREVISION   LATESTREADYREVISION                              AGE
+    custom-transformer-collocation   http://custom-transformer-collocation.default.example.com   True           100                              custom-transformer-collocation-predictor-00001   133m
+    ```
+!!! Note
+    If your DNS contains `svc.cluster.local`, then `Inferenceservice` is not exposed through Ingress. you need to [configure DNS](https://knative.dev/docs/install/yaml-install/serving/install-serving-with-yaml/#configure-dns) 
+    or [use a custom domain](https://knative.dev/docs/serving/using-a-custom-domain/) in order to expose the `isvc`.
+
+## Run a prediction
+Prepare the inputs for the inference request. Copy the following Json into a file named `input.json`.
+
+```json
```

Comment:
This is a bit large to add to the page itself, we can link it and allow user to download.

Reply:
done